packages/event-listener/src/listeners/origin.listener.ts

```diff
@@ -204,8 +204,8 @@ export class OriginEventListener implements IOriginEventListener {
             this.conf
         ).sync();
 
-        const matchedDemands = demands.filter(demand => {
-            const { result } = new MatchableDemand(demand).matchesCertificate(
+        const matchedDemands = demands.filter(async demand => {
+            const { result } = await new MatchableDemand(demand).matchesCertificate(
```

Comment:
Is this construct going to work? `await new` looks unfamiliar to me

Reply:
`new` has higher precedence than `await` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#Table